### PR TITLE
Support `ValueTuple<string, object?>` as state of scope.

### PR DESCRIPTION
**Pull request by jimbojim1997**
Change for issue #186.

When calling `ILogger.BeginScope<TState>(TState state)` `TState` can now be a `ValueTuple<string, object?>`, `Item1` is used as the property name and `Item2` is used as the property value.

```csharp
using (_logger.BeginScope(("TransactionId", (object?)12345)) {
    _logger.LogInformation("Transaction completed in {DurationMs}ms...", 30);
}
// Example JSON output:
// {
//	"@t":"2020-10-29T19:05:56.4176816Z",
//	"@m":"Completed in 30ms...",
//	"@i":"51812baa",
//	"DurationMs":30,
//	"SourceContext":"SomeNamespace.SomeService",
//	"TransactionId": 12345
// }
```

### CODE REVIEW

**Comment by nblumhardt** [comment]
Thanks @jimbojim1997! This looks good, even if the need for the second tuple member to be `object?` is a bit awkward, it's an improvement on the dictionary approach and we might open it up to arbitrary tuple types later on :+1: 

8.0.0 is going out in the next ~24 hours so it's too late for this change to make that release, unfortunately. We'll be able to ship it shortly afterwards, though.

File: test/Serilog.Extensions.Logging.Tests/SerilogLoggerScopeTests.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (nblumhardt) [inline test/Serilog.Extensions.Logging.Tests/SerilogLoggerScopeTests.cs:1]: Is the .NET Foundation copyright header intended here, and in the other test file below?

Comment (jimbojim1997) [reply test/Serilog.Extensions.Logging.Tests/SerilogLoggerScopeTests.cs:1]: I don't know what it needs to be so just copied it from another file.

I've had another look and there's a mix of files with this header, `Copyright [year] Serilog Contributors`, and no header. What should I change this and the other file to?

Comment (bartelink) [reply test/Serilog.Extensions.Logging.Tests/SerilogLoggerScopeTests.cs:1]: Standard header as per serilog core without a year in all source files
Optional in test files
(I've yet to do the PR to make it completely consistent in core)
https://github.com/serilog/serilog/pull/1969#issuecomment-1782030648

**Comment by nblumhardt** [comment]
I think the checks for `OriginalFormatPropertyName` can be eliminated on all of the paths except `IEnumerable<...>`, since only a specifically-typed state object will carry these. Given I dropped the ball on this PR over Christmas though, I'll merge now to avoid any busywork around conflicts etc. :-)

Sorry about the long delay @jimbojim1997, thanks for this!

**Pull request merged**